docs/build/build_system.md

```diff
@@ -0,0 +1,71 @@
+# Build System
+
+The scp-machine build system is based on the principles of the [sc-machine](https://github.com/ostis-ai/sc-machine) build system. To learn more about them, read [sc-machine build system documentation](https://ostis-ai.github.io/sc-machine/build/build_system/).
+
+## Compilation steps
+
+This project uses CMake presets. These presets help manage different build configurations, making it easier to work with dependencies and streamline the build process.
+
+If you wish to use CLI instead, you can list all the available configurations in the `CMakePresets.json` file and pick one of the presets:
+
+```sh
+cmake --list-presets
+```
+
+### Configure presets
+
+Configure presets define how the project is configured before building. They specify various settings, including whether to include tests and which dependencies to use.
+
+| **Name**                     | **Display Name**                        | **Description**                                      |
+|------------------------------|-----------------------------------------|------------------------------------------------------|
+| `debug-conan`                | Debug config (Conan)                    | Debug config with tests (Conan dependencies used)    |
+| `release-conan`              | Release config (Conan)                  | Release config (Conan dependencies used)             |
+| `release-with-tests-conan`   | Release config with tests (Conan)       | Release config with tests (Conan dependencies used)  |
+
+## Build presets
+
+Build presets link to the configure presets and specify how to build the project. They determine the output directories based on the selected configuration.
+
+| **Name**   | **Configure Preset**                        |
+|------------|---------------------------------------------|
+| `debug`    | `debug-conan`                               |
+| `release`  | `release-conan`, `release-with-tests-conan` |
+
+The build artifacts will be located in the `build/<Release|Debug>/bin` folder.
```

Comment:
this can be added to the tables above, so it'll be more obvious what is the difference between them